File: Code/turnover.R
## Employees leaving prematurely ##

# Importing the data #
turnover = read.csv('https://raw.githubusercontent.com/iese-bad/DataSci/master/Data/turnover.csv',
  stringsAsFactors=FALSE)
dim(turnover)
head(turnover, 10)
str(turnover)

# Summarizing numeric variables #
summary(turnover[, 1:8])

# Summarizing categorical variables #
tab1 = table(turnover[, 'dept'])
tab1
sort(tab1, decreasing=TRUE)
tab2 = table(turnover[, 'dept'], turnover[, 'salary'])
tab2
tab3 = tapply(turnover[, 'left'], turnover[, 'salary'], mean)
round(tab3, 3)
barplot(tab3, main='Figure 1. Barplot')
tab4 = tapply(turnover[, 'left'], turnover[, 'dept'], mean)
round(sort(tab4, decreasing=TRUE), 3)

# Correlation #
R = cor(turnover[, 1:8])
round(R, 2)
      
# Plotting #
hist(turnover[, 'eval'], main='Figure 2. Histogram', xlab='Evaluation')
plot(eval ~ hours, data=turnover, pch='.', main='Figure 3. Scatterplot',
  xlab='Hours', ylab='Evaluation')

File: Code/turnover.py
## Employees leaving prematurely ##

# Importing the data #
import pandas as pd
turnover = pd.read_csv('https://raw.githubusercontent.com/iese-bad/DataSci/master/Data/turnover.csv')
turnover.shape
turnover.head(10)
turnover.info()

# Summarizing numeric variables #
turnover.iloc[:, 0:8].describe().round(2)

# Summarizing categorical variables #
tab1 = turnover['dept'].value_counts()
tab1
tab2 = pd.crosstab(turnover['dept'], turnover['salary'])
tab2

# Pivot tables #
tab3 = pd.pivot_table(data=turnover, index=['salary'], values=['left'],
    aggfunc='mean')
tab3.round(3)
import matplotlib.pyplot as plt
tab3.plot.bar(title='Figure 1. Barplot', figsize=(7,7), color='0.7')
plt.show()
tab4 = pd.pivot_table(data=turnover, index=['dept'], values=['left'],
    aggfunc='mean',)
tab4.round(3)

# Correlation #
R = turnover.iloc[:, 0:8].corr()
R.round(2)

# Histogram #
plt.figure(1, figsize=(7,7))
plt.hist(turnover['eval'], color='0.7')
plt.title('Figure 2. Histogram')
plt.xlabel('Evaluation')
plt.ylabel('Frequency')
plt.show()

# Scatter plot #
plt.figure(1, figsize=(7,7))
plt.scatter(x=turnover['hours'], y=turnover['eval'], color='0.6', s=1)
plt.title('Figure 3. Scatterplot')
plt.xlabel('Hours')
plt.ylabel('Evaluation')
plt.show()
